src/commands/tag/analyze-missing.command.ts

```diff
@@ -0,0 +1,96 @@
+import { bold, Command } from "../../deps.ts";
+import { CmdGlobalOptions } from "../cmd-options.ts";
+import { setupLogger } from "../../logger.ts";
+import { verifyCliAvailability } from "../../init.ts";
+import { loadConfig } from "../../config/config.model.ts";
+import { MeshAdapterFactory } from "../../mesh/mesh-adapter.factory.ts";
+import { MeshTenant } from "../../mesh/mesh-tenant.model.ts";
+
+export const analyzeMissingTags = new Command()
+  .description(
+    "Analyzes all available tags on tenants and returns the percentage of tenants that make use of this tag.",
+  )
+  .option(
+    "--tags <tags:string[]>",
+    "The list of tags to filter on. If not given, all found tags are considered. Example: --tags Environment,Department",
+  )
+  .option(
+    "--details [details:boolean]",
+    "Shows more details, including which cloud tenants are missing which tag.",
+  )
+  .action(analyzeTagsAction);
+
+interface CmdAnalyzeTagsOptions extends CmdGlobalOptions {
+  tags?: string[];
+  details?: boolean;
+}
+
+async function analyzeTagsAction(options: CmdAnalyzeTagsOptions) {
+  setupLogger(options);
+  await verifyCliAvailability();
+
+  const config = loadConfig();
+  const meshAdapterFactory = new MeshAdapterFactory(config);
+  const meshAdapter = meshAdapterFactory.buildMeshAdapter(options);
+
+  const allTenants = await meshAdapter.getMeshTenants();
+
+  let tagList: string[] = [];
+  if (options.tags) {
+    tagList = options.tags;
+  } else {
+    // https://stackoverflow.com/a/35092559/5976604
+    tagList = [
+      ...new Set(
```

Comment:
d: strange... maybe a own utility function would be helpful to make it more clear what happens here like `makeDistinct(array)` but on the other hand that is also quite a bit of a overkill maybe...

Reply:
Done